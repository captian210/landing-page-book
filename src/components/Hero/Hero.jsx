import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LazyHero from 'react-lazy-hero';
import Particles from 'react-particles-js';
import Container from '../Container/Container';
import Button from '../Button/Button';
import Header from '../Header/Header';
// import StarMeUpTemplate from './templates/StarMeUpTemplate';
import { crazyStars } from './particlesSugarMapping';

class Hero extends Component {
  static onButtonClick(button) {
    if (button.onClick && typeof button.onClick === 'function') button.onClick({ button });
  }

  static getCustomCssFilterForImg(props, state) {
    const { imgFilter } = props;
    const { randomClass } = state;
    return `
    .Hero.${randomClass} [src] {
      filter: ${imgFilter};
    }`;
  }

  static getParticlesParams(particles, particlesSugar, particlesParams) {
    if (particlesParams) {
      return particlesParams;
    }
    if (!particlesParams && particlesSugar && particlesSugar === 'crazyStars') {
      return crazyStars;
    }
    return {};
  }

  constructor(props) {
    super(props);
    this.state = {
      randomClass: `Hero_${new Date().getTime()}_${Math.random()
        .toString()
        .replace('.', '')}`
    };
  }
  // static getTemplate(template) {
  //   if(template && template === 'starMeUp'){
  //     return StarMeUpTemplate;
  //   }
  //   return {};
  // }

  render() {
    const {
      image,
      parallaxOffset,
      isFixed,
      isCentered,
      overlayColor,
      opacity,
      header,
      subHeader,
      minHeight,
      subHeaderPosition,
      particles,
      particlesParams,
      particlesSugar,
      backgroundColor,
      buttons,
      children,
      style,
      imgFilter
    } = this.props;

    const { randomClass } = this.state;

    const customCssFilter = Hero.getCustomCssFilterForImg(this.props, this.state);
    console.log(customCssFilter);
    const particlesSugarParams = Hero.getParticlesParams(
      particles,
      particlesSugar,
      particlesParams
    );

    // const TemplateComponent = Hero.getTemplate(template);

    return (
      <div className={`Hero ${randomClass}`} style={{ backgroundColor }}>
        <LazyHero
          isFixed={isFixed}
          isCentered={isCentered}
          imageSrc={image}
          opacity={opacity}
          color={overlayColor}
          parallaxOffset={parallaxOffset}
          minHeight={minHeight}
          style={style}
        >
          {particles && (
            <div className="Hero__Particles">
              <Particles params={particlesSugarParams} />
            </div>
          )}
          <div className="Hero__Container d-flex align-content-center">
            {/* {!children && template && (
             <TemplateComponent {...this.props.templateProps} {...this.props}/>
            )} */}
            {!children && (
              <Container>
                {subHeader &&
                  subHeaderPosition === 'top' && (
                    <Header textAlign={isCentered && 'center'} className="Hero__SubHeader mb-5">
                      {subHeader}
                    </Header>
                  )}
                {header && (
                  <Header textAlign={isCentered && 'center'} className="Hero__Header mb-5">
                    {header}
                  </Header>
                )}
                {subHeader &&
                  subHeaderPosition !== 'top' && (
                    <Header
                      textAlign={isCentered && 'center'}
                      type="h2"
                      className="Hero__SubHeader mb-5"
                    >
                      {subHeader}
                    </Header>
                  )}
                {buttons.map(button => (
                  <Button
                    className="btn-xl Hero__Button"
                    key={button.text}
                    onClick={() => Hero.onButtonClick(button)}
                  >
                    {button.text}
                  </Button>
                ))}
              </Container>
            )}
            {children}
          </div>
          {imgFilter && <style>{customCssFilter}</style>}
        </LazyHero>
      </div>
    );
  }
}

Hero.propTypes = {
  isFixed: PropTypes.bool,
  isCentered: PropTypes.bool,
  particles: PropTypes.bool,
  opacity: PropTypes.number,
  parallaxOffset: PropTypes.number,
  image: PropTypes.string,
  overlayColor: PropTypes.string,
  header: PropTypes.string,
  subHeader: PropTypes.string,
  subHeaderPosition: PropTypes.string,
  minHeight: PropTypes.string,
  backgroundColor: PropTypes.string,
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      onClick: PropTypes.func
    })
  ),
  particlesParams: PropTypes.objectOf(PropTypes.any),
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  style: PropTypes.string,
  imgFilter: PropTypes.string,
  particlesSugar: PropTypes.string
};
Hero.defaultProps = {
  isFixed: true,
  isCentered: true,
  particles: true,
  opacity: 0,
  parallaxOffset: 0,
  image: '',
  overlayColor: '#fff',
  header: '',
  subHeader: '',
  subHeaderPosition: 'bottom',
  minHeight: '100vh',
  backgroundColor: 'transparent',
  buttons: [],
  particlesParams: null,
  children: null,
  style: null,
  imgFilter: null,
  particlesSugar: null
};

export default Hero;
