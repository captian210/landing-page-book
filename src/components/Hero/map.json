{"description":"","displayName":"Hero","methods":[{"name":"onButtonClick","docblock":null,"modifiers":["static"],"params":[{"name":"button","type":null}],"returns":null},{"name":"getCustomCssFilterForImg","docblock":null,"modifiers":["static"],"params":[{"name":"props","type":null},{"name":"state","type":null}],"returns":null},{"name":"getParticlesParams","docblock":null,"modifiers":["static"],"params":[{"name":"particles","type":null},{"name":"particlesSugar","type":null},{"name":"particlesParams","type":null}],"returns":null}],"props":{"isFixed":{"type":{"name":"bool"},"required":false,"description":"Define if background image is fixed. Default `true`","defaultValue":{"value":"true","computed":false}},"isCentered":{"type":{"name":"bool"},"required":false,"description":"Define if content is centered. Default `true`","defaultValue":{"value":"true","computed":false}},"particles":{"type":{"name":"bool"},"required":false,"description":"Define if component show particles animations. Default `true`","defaultValue":{"value":"true","computed":false}},"opacity":{"type":{"name":"number"},"required":false,"description":"Opacity level. Default `0`","defaultValue":{"value":"0","computed":false}},"parallaxOffset":{"type":{"name":"number"},"required":false,"description":"Offset of background image. Default `0`","defaultValue":{"value":"0","computed":false}},"image":{"type":{"name":"string"},"required":false,"description":"Url of background image. Default `\"\"`","defaultValue":{"value":"''","computed":false}},"overlayColor":{"type":{"name":"string"},"required":false,"description":"Overlay color of the component . Default `#fff`","defaultValue":{"value":"'#fff'","computed":false}},"header":{"type":{"name":"string"},"required":false,"description":"Header of the hero . Default `\"\"`","defaultValue":{"value":"''","computed":false}},"subHeader":{"type":{"name":"string"},"required":false,"description":"Sub header of the hero . Default `\"\"`","defaultValue":{"value":"''","computed":false}},"subHeaderPosition":{"type":{"name":"string"},"required":false,"description":"Position of hero sub header. Default `\"bottom\"`, can be `\"top\"`","defaultValue":{"value":"'bottom'","computed":false}},"minHeight":{"type":{"name":"string"},"required":false,"description":"Min height of hero. Default `\"100vh\"`","defaultValue":{"value":"'100vh'","computed":false}},"backgroundColor":{"type":{"name":"string"},"required":false,"description":"Background color of the hero. Default `\"transparent\"`","defaultValue":{"value":"'transparent'","computed":false}},"headerMarginBottom":{"type":{"name":"string"},"required":false,"description":"Margin bottom of hero header. Default `\"5\"`","defaultValue":{"value":"'5'","computed":false}},"headerTextColor":{"type":{"name":"string"},"required":false,"description":"Color of hero header. Default `\"light\"`","defaultValue":{"value":"'light'","computed":false}},"subHeaderMarginBottom":{"type":{"name":"string"},"required":false,"description":"Margin bottom of hero subheader. Default `\"5\"`","defaultValue":{"value":"'5'","computed":false}},"subHeaderTextColor":{"type":{"name":"string"},"required":false,"description":"Color of hero subheader. Default `\"light\"`","defaultValue":{"value":"'light'","computed":false}},"callToAction":{"type":{"name":"string"},"required":false,"description":"Array with buttons to show in hero. Default `[]`, each element must be a object with this structure: `{text: \"TELL ME MORE\", onClick: ({button})=>{}}`","defaultValue":{"value":"null","computed":false}},"particlesParams":{"type":{"name":"objectOf","value":{"name":"any"}},"required":false,"description":"Config params for particles animation.","defaultValue":{"value":"null","computed":false}},"children":{"type":{"name":"union","value":[{"name":"arrayOf","value":{"name":"node"}},{"name":"node"}]},"required":false,"description":"Elements to show inside of Hero.","defaultValue":{"value":"null","computed":false}},"style":{"type":{"name":"string"},"required":false,"description":"Customs styles to tramsform your Hero","defaultValue":{"value":"null","computed":false}},"imgFilter":{"type":{"name":"string"},"required":false,"description":"The filter property defines visual effects (like blur and saturation) to an element (often <img>).","defaultValue":{"value":"null","computed":false}},"particlesSugar":{"type":{"name":"string"},"required":false,"description":"Preconfigured params for particles, you can find this on particlesSugarMapping.js file.","defaultValue":{"value":"null","computed":false}},"callToActionOnClick":{"type":{"name":"func"},"required":false,"description":"","defaultValue":{"value":"() => {}","computed":false}},"callToActionColor":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"''","computed":false}},"callToActionOutline":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"secondaryCallToActionOnClick":{"type":{"name":"func"},"required":false,"description":"","defaultValue":{"value":"() => {}","computed":false}},"secondaryCallToActionColor":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"''","computed":false}},"secondaryCallToActionOutline":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}}},"filename":"/components/Hero/Hero.jsx"}