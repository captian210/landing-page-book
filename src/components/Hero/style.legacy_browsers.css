:root {
  --Hero__Header-fontSize: 75px;
  --Hero__Header-fontStyle: normal;
  --Hero__Header-color: var(#fff);
  --Hero__Header-fontWeight: 700;
  --Hero__Header--phone-fontSize: 30px;
  --Hero__SubHeader-fontSize: 40px;
  --Hero__SubHeader-color: var(#fff);
  --Hero__SubHeader-fontStyle: italic;
  --Hero__SubHeader--phone-fontSize: 20px;
  --Hero__Button-color: var(#fff);
  --Hero__Button-backgroundColor: #fed136; }

:root {
  --Button-borderRadius: 0px;
  --Button-fontSize: 1rem;
  --Button-padding: 0.375rem 0.75rem; }

.Button {
  border-radius: 0px;
  font-size: 1rem;
  padding: 0.375rem 0.75rem; }

.Hero .Hero__Header {
  font-weight: 700;
  font-size: 75px;
  font-style: normal;
  color: var(#fff); }

.Hero .Hero__SubHeader {
  font-size: 40px;
  font-style: italic;
  color: var(#fff); }

.Hero .Hero__Button {
  padding: 20px 40px;
  color: var(#fff);
  background-color: #fed136; }

.Hero .Hero__Particles {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  z-index: 1; }
  .Hero .Hero__Particles div {
    height: 100%; }

.Hero .Hero__Container {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 2; }
  .Hero .Hero__Container .container {
    margin: auto; }

@media (max-width: 768px) {
  .Hero__Header {
    font-size: 30px !important; }
  .Hero__SubHeader {
    font-size: 20px !important; } }
