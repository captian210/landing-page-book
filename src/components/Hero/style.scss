:root {
    --Hero__Header-fontSize: 75px;
    --Hero__Header--phone-fontSize: 30px;
    --Hero__SubHeader-fontSize: 40px;
    --Hero__SubHeader--phone-fontSize: 20px;
}

@import '../Container/style.scss';
@import '../Button/style.scss';
@import '../Header/style.scss';
.Hero {
    .Hero__Header {
        font-size: var(--Hero__Header-fontSize);
    }
    .Hero__SubHeader {
        font-size: var(--Hero__SubHeader-fontSize);
    }
    .Hero__Particles {
        position: absolute;
        width: 100%;
        height: 100%;
        left: 0;
        top: 0;
        z-index: 1;
        div {
            height: 100%;
        }
    }
    .Hero__Container {
        position: absolute;
        width: 100%;
        height: 100%;
        top: 0;
        left: 0;
        z-index: 2;
        .container {
            margin: auto;
        }
    }
}

@media (max-width: 768px) {
    .Hero__Header {
        font-size: var(--Hero__Header--phone-fontSize) !important;
    }
    .Hero__SubHeader {
        font-size: var(--Hero__SubHeader--phone-fontSize) !important;
    }
}