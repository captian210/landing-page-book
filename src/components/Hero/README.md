# Hero

[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

## Overview

`@front10/landing-page-book/Hero` is used to create cool heroes components.

## How to use

1- Importing bootstrap style

```js
import 'bootstrap/dist/css/bootstrap.min.css';
```

2- Add following content to import the component:

```js
import Hero from '@front10/landing-page-book/dist/components/Hero';
import '@front10/landing-page-book/dist/components/Hero/style.css';
```

> Note: For including all components styles once you can use [a theme](https://github.com/front10/landing-page-book/wiki/Theming).

3- Put this code into jsx page:

```html
<Hero
  buttons={[
    {
      onClick: function noRefCheck() {},
      text: 'TELL ME MORE'
    }
  ]}
  header="REACT COMPONENTS"
  opacity={1}
  overlayColor="#373D45"
  particlesParams={{
    move: {
      speed: 2
    }
  }}
  subHeader="Leverage your process!"
/>
```
[![Edit Hero demo](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/yw19mmpl99)

You can use Hero component with children too:

```html
<Hero>
    <Container>
        <Image className="w-75"
             src="http://www.pngpix.com/wp-content/uploads/2016/06/PNGPIX-COM-Aston-Martin-White-Car-PNG-Image.png"/>
        <h1 className="display-3">GET YOUR CAR</h1>
        <div>
            <Button className="btn-success btn-lg">
                Go now
            </Button>
        </div>
        <div className="btn-group mt-4" role="group" aria-label="Basic example">
            <Button className="btn-secondary"><Icon icon="fa fa-facebook"/></Button>
            <Button className="btn-secondary"><Icon icon="fa fa-instagram"/></Button>
            <Button className="btn-secondary"><Icon icon="fa fa-google-plus"/></Button>
        </div>
    </Container>
</Hero>
```
[![Edit Hero with childrens demo](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/yqqorl4qr1)

Note: if you use Hero component with children, properties `buttons`, `header`, `subHeader` and `subHeaderPosition` will not work

## Properties

| </br>Name         | </br>Type | </br>Summary                                                                                                                                           |
| ----------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------ |
| isFixed           | `Boolean` | Define if background image is fixed. Default `true`                                                                                                    |
| isCentered        | `Boolean` | Define if content is centered. Default `true`                                                                                                          |
| particles         | `Boolean` | Define if component show particles animations. Default `true`                                                                                          |
| opacity           | `Number`  | Opacity level. Default `0`                                                                                                                             |
| parallaxOffset    | `Number`  | Offset of background image. Default `0`                                                                                                                |
| image             | `String`  | Url of background image. Default `""`                                                                                                                  |
| overlayColor      | `String`  | Overlay color of the component . Default `#fff`                                                                                                        |
| header            | `String`  | Header of the hero . Default `""`                                                                                                                      |
| subHeader         | `String`  | Sub header of the hero . Default `""`                                                                                                                  |
| subHeaderPosition | `String`  | Position of hero sub header. Default `"bottom"`, can be `"top"`                                                                                        |
| minHeight         | `String`  | Min height of hero. Default `"100vh"`                                                                                                                  |
| backgroundColor   | `String`  | Background color of the hero. Default `"transparent"`                                                                                                  |
| particlesParams   | `Object`  | Object with particles config. Default `{}`, yow can see [react-particles-js](https://www.npmjs.com/package/react-particles-js)                         |
| buttons           | `Array`   | Array with buttons to show in hero. Default `[]`, each element must be a object with this structure: `{text: "TELL ME MORE", onClick: ({button})=>{}}` |

## Theming variables

| </br>Name                           | </br>Summary                                  | </br>Default |
| ----------------------------------- | --------------------------------------------- | ------------ |
| --Hero\_\_Header-fontSize           | Font size of header in hero                   | `75px`       |
| --Hero\_\_Header-fontStyle          | Font style of header in hero                  | `normal`     |
| --Hero\_\_Header-color              | Color of header in hero                       | `#fff`       |
| --Hero\_\_Header-fontWeight         | Font weight of header in hero                 | `700`        |
| --Hero\_\_Header--phone-fontSize    | Font size of header in hero when is mobile    | `30px`       |
| --Hero\_\_SubHeader-fontSize        | Font size of subheader in hero                | `40px`       |
| --Hero\_\_SubHeader-color           | Color of subheader in hero                    | `#fff`       |
| --Hero\_\_SubHeader-fontStyle       | Font style of subheader in hero               | `italic`     |
| --Hero\_\_SubHeader--phone-fontSize | Font size of subheader in hero when is mobile | `20px`       |
| --Hero\_\_Button-color              | Color of buttons hero                         | `#fff`       |
| --Hero\_\_Button-backgroundColor    | Background color of buttons hero              | `#fed136`    |

## Used libraries

- [**react-particles-js**](https://www.npmjs.com/package/react-particles-js)
- [**react-lazy-hero**](http://react-lazy-hero.danielstefanovic.com/)
