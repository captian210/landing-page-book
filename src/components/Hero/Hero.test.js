import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Hero from './Hero';

configure({ adapter: new Adapter() });

describe('Hero', () => {
  let props;
  let mounted;
  const hero = () => {
    if (!mounted) {
      mounted = shallow(<Hero {...props} />);
    }
    return mounted;
  };
  describe('Hero', () => {
    beforeEach(() => {
      props = {
        isFixed: true,
        isCentered: true,
        particles: true,
        opacity: 0,
        parallaxOffset: 0,
        image: '',
        overlayColor: '#fff',
        header: 'AAAAAAAAAAAAAAA',
        subHeader: 'BBBBBBBBBB',
        subHeaderPosition: 'bottom',
        minHeight: '100vh',
        backgroundColor: 'transparent',
        buttons: [],
        particlesParams: {},
        children: null
      };
      mounted = undefined;
    });

    it('should render', () => {
      expect(shallow(<Hero {...props} />)).toMatchSnapshot();
    });

    it('always renders a div', () => {
      const divs = hero().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});
