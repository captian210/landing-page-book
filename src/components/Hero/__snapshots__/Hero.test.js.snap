// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Hero should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero
    backgroundColor="transparent"
    buttons={Array []}
    header=""
    image="https://front10.com/img/header-bg.jpg"
    isCentered={true}
    isFixed={true}
    minHeight="100vh"
    opacity={0}
    overlayColor="#fff"
    parallaxOffset={0}
    particles={true}
    particlesParams={Object {}}
    subHeader=""
    subHeaderPosition="bottom"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Hero",
      "style": Object {
        "backgroundColor": "transparent",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Hero",
        "style": Object {
          "backgroundColor": "transparent",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
